File: scripts/regen_products_data.py
#!/usr/bin/env python3
"""
Regenerate public/js/products-data.js from Supabase jewelry_products.
Fetches all rows, maps to frontend format, writes window.PRODUCTS = [...].
"""
import json, os, urllib.request, urllib.parse

SUPABASE_URL = os.environ["SUPABASE_URL"]
SUPABASE_KEY = os.environ["SUPABASE_SERVICE_KEY"]
OUTPUT = os.path.join(os.path.dirname(__file__), "..", "public", "js", "products-data.js")

def fetch_all():
    all_rows = []
    offset = 0
    page = 1000
    while True:
        url = (f"{SUPABASE_URL}/rest/v1/jewelry_products"
               f"?select=id,sku,name,description,category,images,img_primary,img_hover,min_price,max_price,variations,active"
               f"&order=id.asc"
               f"&limit={page}&offset={offset}")
        req = urllib.request.Request(url)
        req.add_header("apikey", SUPABASE_KEY)
        req.add_header("Authorization", f"Bearer {SUPABASE_KEY}")
        with urllib.request.urlopen(req, timeout=30) as resp:
            batch = json.load(resp)
        all_rows.extend(batch)
        if len(batch) < page:
            break
        offset += page
    return all_rows

def transform(row):
    imgs = row.get("images") or []
    if isinstance(imgs, str):
        try: imgs = json.loads(imgs)
        except: imgs = [imgs]
    img1 = row.get("img_primary") or (imgs[0] if imgs else "")
    img2 = row.get("img_hover") or (imgs[1] if len(imgs) > 1 else img1)

    variations_raw = row.get("variations") or []
    if isinstance(variations_raw, str):
        try: variations_raw = json.loads(variations_raw)
        except: variations_raw = []

    variacoes = []
    for v in variations_raw:
        price = v.get("price") or 0
        variacoes.append({
            "id":       v.get("id"),
            "desc":     v.get("desc") or "",
            "price":    price,
            "original": price
        })

    return {
        "id":       row["id"],
        "name":     row.get("name") or "",
        "sku":      row.get("sku") or "",
        "cat":      row.get("category") or "OUTROS",
        "imgs":     imgs,
        "img":      img1,
        "img2":     img2,
        "minPrice": row.get("min_price") or 0,
        "maxPrice": row.get("max_price") or 0,
        "variacoes": variacoes,
        "descricao": row.get("description") or ""
    }

print("Fetching from Supabase...")
rows = fetch_all()
print(f"Got {len(rows)} rows")

# Only active products
active = [r for r in rows if r.get("active") is not False]
print(f"Active: {len(active)}")

products = [transform(r) for r in active]

out_path = os.path.abspath(OUTPUT)
with open(out_path, "w", encoding="utf-8") as f:
    f.write("/* Lagos Jewelry — Full Product Data (shared by /jewelry and /admin) */\n")
    f.write("window.PRODUCTS = ")
    json.dump(products, f, ensure_ascii=False, separators=(",", ":"))
    f.write(";\n")

size_kb = os.path.getsize(out_path) / 1024
print(f"Written: {out_path} ({size_kb:.0f}KB, {len(products)} products)")

File: scripts/regen-products-data.mjs
#!/usr/bin/env node
/**
 * regen-products-data.mjs — rebuild public/js/products-data.js from Supabase.
 *
 * Run after every catalog sync or manual DB change. Replaces the brittle
 * inline snippet in CLAUDE.md (which emitted the wrong field name).
 *
 * IMPORTANT field mapping (do not change without updating the frontend):
 *   newArrival <- jewelry_products.new_arrival   (drives ✦NEW badge +
 *                 "Novidades" carousel; frontend reads p.newArrival)
 *   stock      <- stock_qty                      (drives In Stock badge)
 *   featured   <- featured                       (carried through; not yet
 *                 consumed by any frontend)
 *
 * Usage:
 *   export SUPABASE_URL=https://vthtufcomuaiyeussrrj.supabase.co
 *   export SUPABASE_SERVICE_KEY=...
 *   node scripts/regen-products-data.mjs
 */

import fs from 'fs';

const BASE = process.env.SUPABASE_URL;
const KEY = process.env.SUPABASE_SERVICE_KEY;
if (!BASE || !KEY) {
  console.error('Missing SUPABASE_URL or SUPABASE_SERVICE_KEY env vars.');
  process.exit(1);
}
const h = { apikey: KEY, Authorization: 'Bearer ' + KEY };

const sel = [
  'id', 'source_id', 'sku', 'name', 'description', 'category',
  'images', 'img_primary', 'img_hover', 'min_price', 'max_price',
  'variations', 'active', 'featured', 'new_arrival', 'stock_qty', 'created_at',
].join(',');

const r = await fetch(
  `${BASE}/rest/v1/jewelry_products?select=${sel}&active=eq.true&order=category.asc,name.asc&limit=3000`,
  { headers: h }
);
if (!r.ok) { console.error('DB fetch failed:', r.status, await r.text()); process.exit(1); }
const products = await r.json();

const t = products.map((p) => {
  const imgs = Array.isArray(p.images) ? p.images : [];
  const v = (p.variations || []).map((x) => ({
    id: x.id, desc: x.desc || '',
    price: x.price || p.min_price || 0,
    original: x.price || p.min_price || 0,
  }));
  return {
    id: p.id,
    name: p.name || '',
    sku: p.sku || '',
    cat: p.category || 'OUTROS',
    imgs,
    img: p.img_primary || imgs[0] || '',
    img2: p.img_hover || imgs[1] || imgs[0] || '',
    minPrice: p.min_price || 0,
    maxPrice: p.max_price || p.min_price || 0,
    variacoes: v,
    descricao: p.description || '',
    featured: !!p.featured,
    newArrival: !!p.new_arrival,   // <-- the fix: frontend reads p.newArrival
    stock: p.stock_qty || 0,
  };
});

fs.writeFileSync(
  './public/js/products-data.js',
  '/* Lagos Jewelry — Full Product Data */\n' +
  '/* AUTO-GENERATED ' + new Date().toISOString() + ' — scripts/regen-products-data.mjs */\n' +
  'window.PRODUCTS=' + JSON.stringify(t) + ';'
);

const stats = {
  products: t.length,
  newArrival: t.filter((x) => x.newArrival).length,
  inStock: t.filter((x) => x.stock > 0).length,
  featured: t.filter((x) => x.featured).length,
};
console.log('Done:', JSON.stringify(stats));
